database/index.js

```diff
@@ -0,0 +1,81 @@
+// create a connection
+const mongoose = require('mongoose');
+
+// import db credentials
+const config = require('../config.js');
+
+mongoose.connect(`mongodb://${config.DB_ID}:${config.DB_PASSWORD}@ds141889.mlab.com:41889/booking`);
+
+// To connect the db in the shell
+// mongo ds141889.mlab.com:41889/booking -u <dbuser> -p <dbpassword>
+
+const db = mongoose.connection;
+db.on('error', console.error.bind(console, 'connection error:'));
+db.once('open', () => {
+  console.log('DB Connected!');
+});
+
+// create a schema for DB data
+const bookingSchema = mongoose.Schema({
+
+  room_id: Number,
+  room_name: String,
+  world_name: String,
+  keywords: String,
+  room_rate: Number,
+  booked_dates: [String], // store booked dates in an array
+  // in the future, booked_dates can be [checkin, checkout, guestnumber]
+  guest_number: Number,
+  guest_name: String,
+  host_name: String,
+  discount: Boolean,
+  cleaning_fee: Boolean,
+  review_count: Number,
+  review_grade: Number,
+  created_date: { type: Date, default: Date.now },
+
+});
+
+// create a model for the schema
+const Room = mongoose.model('room', bookingSchema);
+
+
+// adding booking dates to DB (only booked_dates)
+const update = (data) => {
+
+  let newBooking = {
+    room_id: data.id,
+    booked_dates: data.booked,
+    // guest_number: data.guest_number,
+    guest_name: data.guest_name,
+  };
+
+  Room.findOneAndUpdate({ room_id: data.id }, { $push: { booked_dates: data.booked } }, (err, room) => {
+    if (err) return console.error(err);
```

Comment:
this is probably the best place to have the callback from the server

Reply:
@jennqiao  Do you mean I need a callback to confirm the update from the server-side?
This query is just to insert a new booking date so I'm not sure if the callback is necessary.
Would you mind checking it again? Thanks!

Reply:
@jennqiao Actually I'll add a callback to handle an error here. I saw your comment below. Thanks :)